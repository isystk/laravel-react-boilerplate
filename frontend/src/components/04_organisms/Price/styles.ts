import { css, keyframes } from '@emotion/css'

export const price = css``
