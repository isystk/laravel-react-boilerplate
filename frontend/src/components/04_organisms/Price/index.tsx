import React, { FC } from 'react'
import { ContainerProps } from 'types'
import * as styles from './styles'
import { connect } from '@/components/hoc'
import LineButton from '@/components/01_atoms/LineButton'
import ScrollIn from '@/components/02_interactions/ScrollIn'
import { useI18n } from '@/components/i18n'

/** PriceProps Props */
export type PriceProps = Record<never, never>
/** Presenter Props */
export type PresenterProps = PriceProps & {
  t
}

/** Presenter Component */
const PricePresenter: FC<PresenterProps> = ({ t, ...props }) => (
  <>
    <section className={`${styles.price}`}>
      <div className="flex items-center bg-black justify-center w-full">
        <div className="py-32 w-full">
          <ScrollIn>
            <p className="text-white font-bold text-3xl md:mb-16 text-center">
              PRICE
            </p>
          </ScrollIn>
          <div className="flex flex-wrap items-center p-8 md:mb-16">
            <div className="flex w-full md:w-1/2 justify-end items-center">
              <ScrollIn className="w-full md:w-72 md:mr-8">
                <div className="bg-white bg-opacity-20 rounded-lg py-16 h-64 mb-8 md:mb-0">
                  <p className="text-white text-center mb-8">
                    {t('Limited number of chats')}
                  </p>
                  <p className="text-white font-bold text-center text-5xl">
                    {t('free')}
                  </p>
                </div>
              </ScrollIn>
            </div>
            <div className="flex w-full md:w-1/2 justify-start items-center">
              <ScrollIn className="w-full md:w-72 md:ml-8">
                <div className="bg-white bg-opacity-20 rounded-lg py-16 h-64 ">
                  <p className="text-white text-center mb-8">
                    {t('Unlimited Chat')}
                  </p>
                  <div className="px-8">
                    <p className="text-white font-bold text-left text-4xl">
                      <span className="text-white text-xl mr-4">
                        {t('monthly amount')}
                      </span>
                      ¥980
                    </p>
                    <p className="text-white font-bold text-left text-4xl">
                      <span className="text-white text-xl mr-4">
                        {t('yearly amount')}
                      </span>
                      ¥9,800
                    </p>
                  </div>
                </div>
              </ScrollIn>
            </div>
          </div>
          <div className="flex items-center justify-center px-8">
            <LineButton
              link="#"
              label={t('Add me as a friend and ask me a question')}
            />
          </div>
        </div>
      </div>
    </section>
  </>
)

/** Container Component */
const PriceContainer: React.FC<ContainerProps<PriceProps, PresenterProps>> = ({
  presenter,
  ...props
}) => {
  const { t } = useI18n('Common')

  return presenter({
    t,
    ...props,
  })
}

export default connect<PriceProps, PresenterProps>(
  'Price',
  PricePresenter,
  PriceContainer
)
