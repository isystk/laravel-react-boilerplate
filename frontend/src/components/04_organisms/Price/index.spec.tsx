import React from 'react'
import renderer from 'react-test-renderer'
import Price from './index'

describe('Price', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(<Price />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
