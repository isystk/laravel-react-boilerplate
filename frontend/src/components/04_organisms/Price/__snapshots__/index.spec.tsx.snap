// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Match Snapshot 1`] = `
<section
  className="css-0"
>
  <div
    className="flex items-center bg-black justify-center w-full"
  >
    <div
      className="py-32 w-full"
    >
      <div
        className="undefined css-pjt9eq css-1exhw58"
      >
        <p
          className="text-white font-bold text-3xl md:mb-16 text-center"
        >
          PRICE
        </p>
      </div>
      <div
        className="flex flex-wrap items-center p-8 md:mb-16"
      >
        <div
          className="flex w-full md:w-1/2 justify-end items-center"
        >
          <div
            className="w-full md:w-72 md:mr-8 css-pjt9eq css-1exhw58"
          >
            <div
              className="bg-white bg-opacity-20 rounded-lg py-16 h-64 mb-8 md:mb-0"
            >
              <p
                className="text-white text-center mb-8"
              >
                Limited number of chats
              </p>
              <p
                className="text-white font-bold text-center text-5xl"
              >
                free
              </p>
            </div>
          </div>
        </div>
        <div
          className="flex w-full md:w-1/2 justify-start items-center"
        >
          <div
            className="w-full md:w-72 md:ml-8 css-pjt9eq css-1exhw58"
          >
            <div
              className="bg-white bg-opacity-20 rounded-lg py-16 h-64 "
            >
              <p
                className="text-white text-center mb-8"
              >
                Unlimited Chat
              </p>
              <div
                className="px-8"
              >
                <p
                  className="text-white font-bold text-left text-4xl"
                >
                  <span
                    className="text-white text-xl mr-4"
                  >
                    monthly
                  </span>
                  ¥980
                </p>
                <p
                  className="text-white font-bold text-left text-4xl"
                >
                  <span
                    className="text-white text-xl mr-4"
                  >
                    yearly
                  </span>
                  ¥9,800
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="flex items-center justify-center px-8"
      >
        <a
          className="css-0 p-2 pl-4 pr-4 flex items-center justify-center bg-accent rounded-full"
          href="#"
        >
          <img
            alt="Add me as a friend and ask me a question"
            height="50px"
            src="/images/icon-line.png"
            width="50px"
          />
          <span
            className="text-white font-bold"
          >
            Add me as a friend and ask me a question
          </span>
        </a>
      </div>
    </div>
  </div>
</section>
`;
